src/hooks/useSupplyWithdrawModal/Modal/SupplyForm/SubmitSection/index.tsx

```diff
@@ -0,0 +1,96 @@
+/** @jsxImportSource @emotion/react */
+import { EnableTokenSteps, PrimaryButton } from 'components';
+import React, { useMemo } from 'react';
+import { useTranslation } from 'translation';
+import { Token } from 'types';
+import { areTokensEqual, getContractAddress } from 'utilities';
+
+import { FormError } from '../useForm/types';
+
+const swapRouterContractAddress = getContractAddress('swapRouter');
+
+export interface SubmitSectionProps {
+  isFormValid: boolean;
+  isFormSubmitting: boolean;
+  toToken: Token;
+  fromToken: Token;
+  fromTokenAmountTokens: string;
+  isSwapLoading: boolean;
+  formError?: FormError;
+}
+
+export const SubmitSection: React.FC<SubmitSectionProps> = ({
+  isFormValid,
+  isFormSubmitting,
+  toToken,
+  fromToken,
+  fromTokenAmountTokens,
+  formError,
+  isSwapLoading,
+}) => {
+  const { t } = useTranslation();
+
+  const submitButtonLabel = useMemo(() => {
+    if (isSwapLoading && Number(fromTokenAmountTokens) > 0) {
+      return t('supplyWithdrawModal.supply.submitButtonLabel.processing');
+    }
+
+    if (formError === 'SUPPLY_CAP_ALREADY_REACHED') {
+      return t('supplyWithdrawModal.supply.submitButtonLabel.supplyCapReached');
+    }
+
+    if (formError === 'SWAP_INSUFFICIENT_LIQUIDITY') {
+      return t('supplyWithdrawModal.supply.submitButtonLabel.insufficientSwapLiquidity');
+    }
+
+    if (formError === 'SWAP_WRAPPING_UNSUPPORTED') {
+      return t('supplyWithdrawModal.supply.submitButtonLabel.wrappingUnsupported');
+    }
+
+    if (formError === 'SWAP_UNWRAPPING_UNSUPPORTED') {
+      return t('supplyWithdrawModal.supply.submitButtonLabel.unwrappingUnsupported');
+    }
+
+    if (formError === 'HIGHER_THAN_SUPPLY_CAP') {
+      return t('supplyWithdrawModal.supply.submitButtonLabel.amountHigherThanSupplyCap');
+    }
+
+    if (formError === 'HIGHER_THAN_WALLET_BALANCE') {
+      return t('supplyWithdrawModal.supply.submitButtonLabel.insufficientWalletBalance', {
+        tokenSymbol: fromToken.symbol,
+      });
+    }
```

Comment:
nit: I feel this is better expressed as a `switch(formError) case...`

Reply:
The issue is that `formError` isn't the only variables used to determine what to return here: some of the IF statements use `isSwapLoading`, `fromTokenAmountTokens` and `isFormValid`. That's why I'd keep the IF statements, otherwise in order to use a switch we'll have to make a separate function